rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSourceWindow.java

```diff
@@ -0,0 +1,452 @@
+        HiddenItem deploy = new HiddenItem("deploy");
+        HiddenItem nodeSourceEdited = new HiddenItem("nodeSourceEdited");
+        HiddenItem callback = new HiddenItem("nsCallback");
+        HiddenItem session = new HiddenItem("sessionId");
+
+        ArrayList<FormItem> formItems = new ArrayList<>();
+
+        formItems.add(name);
+        formItems.add(nodesRecoverable);
+        formItems.add(deploy);
+        formItems.add(nodeSourceEdited);
+        formItems.add(callback);
+        formItems.add(session);
+
+        return formItems;
+    }
+
+    protected ArrayList<FormItem> getPrefilledFormItems(PluginDescriptor plugin) {
+
+        List<PluginDescriptor.Field> pluginFields = plugin.getConfigurableFields();
+        ArrayList<FormItem> formItems = new ArrayList<>(pluginFields.size());
+
+        FormItem formItem;
+        for (PluginDescriptor.Field pluginField : pluginFields) {
+
+            if (pluginField.isPassword()) {
+
+                formItem = new PasswordItem(plugin.getPluginName() + pluginField.getName(), pluginField.getName());
+
+            } else if (pluginField.isFile() || pluginField.isCredential()) {
+
+                formItem = new UploadItem(plugin.getPluginName() + pluginField.getName(), pluginField.getName());
+
+                if (pluginField.isCredential()) {
+
+                    PickerIcon createCredentialsPicker = new PickerIcon(new PickerIcon.Picker(Images.instance.key_16()
+                                                                                                             .getSafeUri()
+                                                                                                             .asString()),
+                                                                        formItemIconClickEvent -> {
+                                                                            CredentialsWindow win = new CredentialsWindow();
+                                                                            win.show();
+                                                                        });
+                    createCredentialsPicker.setPrompt("Create a Credential file");
+                    createCredentialsPicker.setWidth(16);
+                    createCredentialsPicker.setHeight(16);
+                    createCredentialsPicker.setAttribute("hspace", 6);
+
+                    formItem.setIcons(createCredentialsPicker);
+                }
+            } else {
+
+                formItem = new TextItem(plugin.getPluginName() + pluginField.getName(), pluginField.getName());
+            }
+
+            formItem.setValue(pluginField.getValue());
+            formItem.setWidth(250);
+            formItem.setHint("<nobr>" + pluginField.getDescription() + "</nobr>");
+
+            formItems.add(formItem);
+        }
+        return formItems;
+    }
+
+    // TODO Method to use when the server is ready to deal with
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)